python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,974 @@
+                        elif i.debugName() in self._inputs_r.keys():
+                            listconstr.append(int(self._inputs_r[i.debugName()]))
+
+                    # Unwrap for tensors
+                    if len(listconstr) == 1:
+                        listconstr = listconstr[0]
+
+                    self._relay_map[nid] = listconstr
+                    self._nid_to_node_name[op_name] = nid
+                    nid = nid+1
+            else:
+                for i in op_node.inputs():
+                    if i.debugName() in self._nid_to_node_name.keys():
+                        for cnt in range(0, len(self._op_inputs_r[(op_name, operator)])):
+                            if isinstance(self._op_inputs_r[(op_name, operator)][cnt], str):
+                                if "call/var" in self._op_inputs_r[(op_name, operator)][cnt]:
+                                    self._op_inputs_r[(op_name, operator)][cnt] = \
+                                        self._relay_map[self._nid_to_node_name[i.debugName()]]
+                                    break
+
+                call = _convert_map[operator](self._op_inputs_r[(op_name, operator)])
+
+                self._relay_map[nid] = call
+                self._nid_to_node_name[op_name] = nid
+                nid = nid+1
+
+        outputs = []
+
+        for i in range(nid):
+            output = self._relay_map[i]
+            outputs.append(output)
+
+        if len(outputs) == 1:
+            body = outputs[0]
+        else:
+            body = outputs[-1]
+
+        func = tvm.relay.Function(self._fn_param, body)
+
+        param = {k: tvm.nd.array(v) for k, v in self._params.items()}
+
+        return  _module.Module.from_expr(func), param
+
+    def _parse_inputs(self):
+        """ Map inputs to parser and inputs to graph. """
+        # Get names and objects of inputs for IR
+        ir_names = [i.debugName() for i in self._trace.graph.inputs()]
+        ir_inputs = [i for i in self._trace.graph.inputs()]
+
+        # Create corresponding shape and add to input
+        for input_name, ir_input in zip(self._input_shapes, ir_inputs[1:]):
+            input_shape = self._input_shapes[input_name]
+            tensor = tvm.nd.array(np.zeros(input_shape).astype(np.float32))
+            ir_input.setDebugName(input_name)
+            self._inputs_r[input_name] = _expr.var(input_name,
+                                                   shape=self._input_shapes[input_name],
+                                                   dtype='float32')
+            self._fn_param.append(_expr.var(input_name,
+                                            shape=self._input_shapes[input_name],
+                                            dtype='float32'))
+
+        # Add self (first input of a PyTorch graph) to inputs
+        input_shape = [3]
+        tensor = tvm.nd.array(np.zeros(input_shape).astype(np.float32))
+        input_name = ir_names[0]
+        self._inputs_r[input_name] = tensor
```

Comment:
It's a reference to an input in the PT graph.

`graph(%self : ClassType<Conv2D1>,
      %input0 : Float(1, 3, 224, 224)):
  %1 : ClassType<Conv2d> = prim::GetAttr[name="conv"](%self)
  %weight : Tensor = prim::GetAttr[name="weight"](%1)
  ...
)`

It is used as an input to the prim:GetAttr operator sometimes but we can hard-code the name 'self' if that seems like a better option. I think it might be better to just add a zero tensor as I am not sure if it is always named 'self'.

Reply:
Ok that makes sense, there's no need for a change in that case.